packages/internet_detector.vm/tools/chocolateyinstall.ps1

```diff
@@ -23,8 +23,7 @@ Copy-Item "$imagesPath\*" ${Env:VM_COMMON_DIR} -Force
 
 VM-Install-Shortcut -toolName $toolName -category $category -executablePath "$toolDir/$toolName.exe"
 
-# TODO - Uncomment when FakeNet BlackList for DNS is fixed/addressed. https://github.com/mandiant/flare-fakenet-ng/issues/190
-# # Create scheduled task for tool to run every 2 minutes.
-# $action = New-ScheduledTaskAction -Execute $rawToolPath
-# $trigger = New-ScheduledTaskTrigger -Once -At (Get-Date) -RepetitionInterval (New-TimeSpan -Minutes 2)
-# Register-ScheduledTask -Action $action -Trigger $trigger -TaskName 'Internet Detector' -Force
+# Create scheduled task for tool to run every 2 minutes.
+$action = New-ScheduledTaskAction -Execute "$toolDir/$toolName.exe"
```

Comment:
is the tool then started several times? :thinking:

Reply:
Hmmm, my testing hadn't shown that, but when I force the Scheduled Task to run manually, it did cause it to create multiple instances.

I think I will have to modify `internet_detector` itself so that it checks if there is one already running before attempting to execute. Doesn't seem like an option we can do with a scheduled task.

Reply:
I see some changes in the internet_detector, but I think I am missing the change for this. @emtuls have you forgotten to add it or am I missing something?

Reply:
There isn't anything that can be done on the scheduled task side that I can see. 🤔

Unless you're still questioning if it should be executed every 2 minutes at all? Which comes down to how we want this tool to be run.

The reason I left that is if the program were to somehow crash or get stopped (either by malware or by an analyst whether it be on accident or not) then this would allow it to start back up without needing any user interaction.

The program itself now checks for if there's already an instance of itself running and if so, it won't run another (of course, the initial execution to determine that will happen, but then it closes).

But if there's a different way you'd like for it to be run, like only once on login/reboot, and any crash/stoppage requiring manual user interaction to start it again, then let me know and I can make that change. 🙂